### PR TITLE
fix: sort query parameter values after encoding

### PR DESCRIPTION
**Pull request by ealain**
This solves the issue #848.

As stated in the [AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/create-signed-request.html#create-canonical-request) on signing API requests (emphasis in mine),
> `CanonicalQueryString` – The URL-encoded query string parameters, separated by ampersands (&). Percent-encode reserved characters, including the space character. Encode names and values separately. If there are empty parameters, append the equals sign to the parameter name before encoding. **After encoding, sort the parameters alphabetically** by key name. If there is no query string, use an empty string ("").

*Description of changes:*

- sort the parameter values after encoding in `getCanonicalQuery`,
- test the associated edge case.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by trivikr — APPROVED** [review]

**Comment by srchase** [comment]
Thanks raising the issue and making this contribution to fix it! Mind running `yarn format` on it?

File: packages/signature-v4/src/getCanonicalQuery.spec.ts [hunk 1]
```diff
@@ -96,6 +96,17 @@ describe("getCanonicalQuery", () => {
     ).toBe("%F0%9F%90%8E=%F0%9F%92%A9&%F0%9F%90%8E=%F0%9F%A6%84");
   });
 
+  it("should sort URI-encode members of query param arrays", () => {
+    expect(
+      getCanonicalQuery(
+        new HttpRequest({
+          ...httpRequestOptions,
+          query: { "p": ["a", "à"] },
```
Comment (srchase) [inline packages/signature-v4/src/getCanonicalQuery.spec.ts:104]: `yarn format` is only complaining about this:
```suggestion
          query: { p: ["a", "à"] },
```
